### ISSUE TITLE
Maintain session-id between requests

### ISSUE THREAD

**Issue by SavchenkoValeriy** [alphapapa/plz.el#25]
SSL handshake can take too long if done every time. That can become pretty painful for API users that need multiple serial requests to implement the core functionality. And in general it would make it much faster to call APIs on the same SSL connection.

`libcurl` can do it, and `curl` does it by default for multiple requests within one invocation. I don't a way to call `curl` multiple times and keep connection alive.

**Comment by alphapapa** [comment]
Thanks for filing this.  Yes, it would be nice to reuse HTTP sessions with curl (e.g. with pipelining), especially to avoid TLS setup.

It may be possible to reuse a curl process to send multiple requests, which would likely solve this.  A possible example of how to do so may be seen in Chris Wellons's `elfeed-curl` library, which inspired `plz`.  See: https://github.com/skeeto/elfeed/blob/162d7d545ed41c27967d108c04aa31f5a61c8e16/elfeed-curl.el#L221-L264

**Comment by alphapapa** [comment]
However, from a cursory search, it doesn't appear that this is likely to work.  See, e.g. https://stackoverflow.com/questions/38857881/curl-command-line-ssl-session-reuse  

And it doesn't sound like it's possible to use a single curl process to start making requests while also continuing to listen for instructions to make more requests.  What is possible is to instruct a single curl process to make multiple requests, and then have it start making them, reusing connections during the sequence; but it sounds like, when the last request is done, the curl process exits, and the TLS session is discarded.

So a partial solution to this would be to enhance plz's queues to use a single curl process when possible, like `elfeed-curl` does.

**Comment by SavchenkoValeriy** [comment]
Yeah, I also read that thread. As a workaround for fellow wanderers, I can I suggest to use ol' trusty `url-retrieve`. It keeps connection alive and for my super tiny benchmark of sending many requests to **httpbin**, it performs ~4 times faster than curl.
